meter/tapo/connection.go

```diff
@@ -292,6 +292,14 @@ func (d *Connection) DoSecureRequest(uri string, taporequest map[string]interfac
 		return nil, err
 	}
 
+	// Login atempt in case of tapo switch connection hicups
+	if res.ErrorCode == 9999 {
+		d.Login()
```

Comment:
Errorhandling?

Reply:
ja, meinst du damit, die Logik doch in die `CheckErrorCode`Funktion` zu verschieben?

Reply:
Ne, ich meine potenzielle Loginfehler auch zu behandeln. Das gibt ja einen err zurück.

Reply:
Aaaah, ok. Hab' ich übersehen ... 

@stefanknotzer du müsstest das Errorhandling für den Login aufruf noch mit aufnehmen:

```go
...
	// Second login atempt in case of tapo switch login hicups
	if res.ErrorCode == 9999 {
		if err := d.Login(); err != nil {
			return nil, err
		}

		if err := d.DoJSON(req, &res); err != nil {
			return nil, err
		}
	}
```